### ISSUE TITLE
Timeslice Issue

### ISSUE THREAD

**Issue by code-wizard** [collab-project/videojs-record#376]
Hey, Am having issues with the timestamp event. I am trying to upload a "slice" of the recorded video. Below is what I noticed.

1. The first slice of the `player.recordedData` contains a blob of size 1

2. The subsequent slices contain blob array of size + 1

I did the following
`var binaryData = player.recordedData[player.recordedData.length - 1];`
The first blob is playable but the subsequent blobs are not playable.
I only want to send a particular slice at a time and not the total recorded file. How do I achieve this?

**Comment by code-wizard** [comment]
I manage to get this working.

**Comment by thijstriemstra** [comment]
@code-wizard opening a ticket and then closing it without sharing the solution.. how did you get it to work?

**Comment by saswatkumarsahoo** [comment]
Hi, I am facing the same issue. The first slice is only playable. Rest of the slices seems to be corrupt. 

@code-wizard  - can you let me know how did you get this working ? 

My code `  var options = {
            controls: true,
            width: 320,
            height: 240,
            fluid: false,
            videoMimeType: 'codecs=H264',
            plugins: {
                record: {
                    audio: true,
                    video: true,
                    maxLength: 60,
                    debug: true,
                    timeSlice: 5000
                }
            }
        };`

` player.on('timestamp', function () {
            console.log(JSON.stringify(player.recordedData))
            if (player.recordedData && player.recordedData.length > 0) {
                var binaryData = player.recordedData[player.recordedData.length - 1];
                 upload(binaryData);
            }
        });`

`function upload(blob) {
            var serverUrl = '/upload';
            var formData = new FormData();
            formData.append('file', blob, blob.name);
            console.log('upload recording ' + blob.name + ' to ' + serverUrl);
            fetch(serverUrl, {
                method: 'POST',
                body: formData
            }).then(
                success => console.log('upload recording complete.'+ formData)
            ).catch(
                error => console.error('an upload error occurred!')
            );
        }`

**Comment by code-wizard** [comment]
> @code-wizard opening a ticket and then closing it without sharing the solution.. 👎 how did you get it to work?
That's too bad of me!
 What I did was I sent the segments to my s3 bucket then used Open CV to stitch them together. So basically the blobs except the 1st segment cannot play independently unless stitched together.

**Comment by code-wizard** [comment]
> Hi, I am facing the same issue. The first slice is only playable. Rest of the slices seems to be corrupt.
> 
> @code-wizard - can you let me know how did you get this working ?
> 
> My code ` var options = { controls: true, width: 320, height: 240, fluid: false, videoMimeType: 'codecs=H264', plugins: { record: { audio: true, video: true, maxLength: 60, debug: true, timeSlice: 5000 } } };`
> 
> ` player.on('timestamp', function () { console.log(JSON.stringify(player.recordedData)) if (player.recordedData && player.recordedData.length > 0) { var binaryData = player.recordedData[player.recordedData.length - 1]; upload(binaryData); } });`
> 
> `function upload(blob) { var serverUrl = '/upload'; var formData = new FormData(); formData.append('file', blob, blob.name); console.log('upload recording ' + blob.name + ' to ' + serverUrl); fetch(serverUrl, { method: 'POST', body: formData }).then( success => console.log('upload recording complete.'+ formData) ).catch( error => console.error('an upload error occurred!') ); }`

I can share my OpenCV function if that will help.

**Comment by saswatkumarsahoo** [comment]
@code-wizard  
Thanks for your response. That will not help actually. My final goal is to upload the video segments to S3 bucket and run rekognition on each slice individually. Stitching them together to create a larger file again won't help. 

Sholudn't the sliced files work independently ? What is the point to slice the recording in the first place if they are not usable. Is it possible to open the ticket again?

**Comment by code-wizard** [comment]
Yeah we should be able to get  playable segments that will be really helpful

**Comment by saswatkumarsahoo** [comment]
Is there any new development on this ?

**Comment by thijstriemstra** [comment]
Don't see any, no. Can you reproduce this with recordrtc only code?

**Comment by thijstriemstra** [comment]
I was able to reproduce the issue with the `timeSlice` option and it looks like a RecordRTC issue: https://github.com/muaz-khan/RecordRTC/issues/539

**Comment by brianarpie** [comment]
@thijstriemstra I'm still having this problem in 3.8.0. What did the merge from July 14 do? What can I do to get past this problem? Thanks

**Comment by thijstriemstra** [comment]
> What did the merge from July 14 do? 

It only fixed an import, related to this issue, but not fixing this issue.

**Comment by anish2591** [comment]
@thijstriemstra Do you know if this has been resolved? I am still seeing this same issue during local testing.

**Comment by anish2591** [comment]
> > Hi, I am facing the same issue. The first slice is only playable. Rest of the slices seems to be corrupt.
> > @code-wizard - can you let me know how did you get this working ?
> 
> I can share my OpenCV function if that will help.

@code-wizard Can you share your OpenCV function? Which library did you use to combine the videos?

**Comment by code-wizard** [comment]
Here is my python code to merge the video files

`for f in tempdir.iterdir():
            if os.stat(str(f)).st_size > 0:
                videofiles.append(str(f))
        video_index = 0
        print(videofiles)
        if len(videofiles)>1:
            cap = cv2.VideoCapture(videofiles[0])

            fourcc = cv2.VideoWriter_fourcc(*'avc1')
            output = "{0}/{1}".format(tempdir, "recorded-output.mp4")
            out = cv2.VideoWriter(output, fourcc, 20.0, (640, 480))

            while (cap.isOpened()):
                ret, frame = cap.read()
                if frame is None:
                    video_index += 1
                    if video_index >= len(videofiles):
                        break
                    cap = cv2.VideoCapture(videofiles[video_index])
                    ret, frame = cap.read()
                out.write(frame)
            cap.release()
            out.release()
            f = open(output,"rb")
            PrivateMediaStorage().save(s3_location,f)`
